modules/tactical/src/socket.ts

```diff
@@ -0,0 +1,203 @@
+   * to identify the data associated with the 'key', the 'id' was provided by Tactical to identify
+   * the 'mutation', and the 'mutation' was provided by the application to be applied to the backend
+   * data model.
+   *
+   * 'resolve' should be called if the 'mutation' can be successfully applied to the data model. A
+   * mutation can only be considered successful if the resulting change to the data model is deeply
+   * equal to the provided 'mutation'. This will notify all connected clients that a new mutation
+   * has been resolved by the backend services.
+   *
+   * 'fail' should be called if the 'mutation' cannot be applied or if the resulting change of
+   * applying the 'mutation' does not deeply equal the 'mutation' provided. This will notify
+   * the calling client that their mutation could not be applied.
+   */
+  onMutation(key: Object, base: string, id: number, mutation: Object, resolve: ResolutionHandler,
+             fail: FailureHandler): void;
+
+  /**
+   * Called by SocketServer when a request is received from the client. The 'key' was provided
+   * by the application to identify the request.
+   *
+   * 'respond' should be called once the request has been satisfied by the backend services.
+   * This will notify the calling client that their request has been satisfied.
+   */
+  onRequest(key: Object, respond: ResponseHandler): void;
+}
+
+/**
+ * A callback result type for ResolutionHandler and ResponseHandler.
+ */
+export interface BaseData {
+  base: string;  // a unique 'base' version provided by the backend service to identify the 'data'
+  data: Object;  // the 'data' to send upstream to the client
+}
+
+/**
+ * A callback result type for FailureHandler.
+ */
+export interface Failure {
+  reason: string;    // a short description of why the mutation was failed
+  context?: Object;  // any 'contextual' data surrounding the 'reason' why the mutation was failed
+}
+
+// Types emitted on socket streams
+//==================================================================================================
+
+/**
+ * The type that is emitted on the 'data' stream.
+ */
+export type DataFrame = VersionedObject;
+
+/**
+ * The type that is emitted on the 'failure' stream.
+ */
+export type FailureFrame = FailedMutation;
+
+/**
+ * The type that is emitted on the 'mutation' stream.
+ */
+export interface MutationFrame {
+  key: Object;       // the 'key' that is associated with the 'mutation'
+  base: string;      // the 'base' that the 'mutation' is targeting
+  id: number;        // the unique 'id' of the 'mutation'
+  mutation: Object;  // the 'mutation' that is to be applied
+  context?: Object;  // contextual information needed to authorize the request
+}
+
+/**
+ * The type that is emitted on the 'request' stream.
+ */
+export interface RequestFrame {
+  key: Object;       // the 'key' that is associated with the request
+  context?: Object;  // contextual information needed to authorize the request
```

Comment:
Context is generic - so change the comment:
`// contextual information associated with this request (for example, authorization information)`